tests/test_cli_client.cpp

```diff
@@ -2393,10 +2399,65 @@ TEST_F(Client, registerCmdHelpOk)
     EXPECT_EQ(send_command({"register", "--help"}), mp::ReturnCode::Ok);
 }
 
-// TODO: This test will change when the echoless passphrase prompt in working
-TEST_F(Client, registerCmdNoPassphrasFails)
+TEST_F(Client, registerCmdAcceptsEnteredPassphrase)
 {
-    EXPECT_EQ(send_command({"register"}), mp::ReturnCode::CommandLineError);
+    const std::string passphrase{"foo"};
+    std::ostringstream cerr, cout;
+    std::istringstream cin;
+    mpt::MockTerminal mock_terminal;
+
+    cin.str(passphrase + "\n");
+
+    EXPECT_CALL(mock_terminal, cout).WillRepeatedly(ReturnRef(cout));
+    EXPECT_CALL(mock_terminal, cerr).WillRepeatedly(ReturnRef(cerr));
+    EXPECT_CALL(mock_terminal, cin).WillRepeatedly(ReturnRef(cin));
+    EXPECT_CALL(mock_terminal, set_cin_echo(false)).Times(1);
+    EXPECT_CALL(mock_terminal, set_cin_echo(true)).Times(1);
```

Comment:
All these could go in a fixture, and true/false should go in a sequence?

Reply:
Yep, all done.